src/stream/src/executor/aggregation/distinct.rs

```diff
@@ -81,7 +81,7 @@ impl<S: StateStore> ColumnDeduplicater<S> {
 
             ctx.streaming_metrics
                 .agg_distinct_total_cache_count
-                .with_label_values(&[&table_id_str, &actor_id_str, &fragment_id_str])
+                .with_guarded_label_values(&[&table_id_str, &actor_id_str, &fragment_id_str])
```

Comment:
We should store the value (counter, gauge and histogram) returned from `with_guarded_label_values` instead of creating it as a temporary variable and then drop it after used.

In current usage, the counter is dropped immediately after `inc`. After the being collected, the counter value will be clear, and the next call on it will reinitialize a new counter with default value 0, which makes the metrics inaccurate.

ditto for all similar usage in this PR.

Reply:
Yeah I just noticed that too...